content/en/docs/dependencies.md

```diff
@@ -110,7 +73,75 @@ This will allow the flexibility to write and generate software where various dis
+{{< /blocks/callout >}}
+
+## Dependency Version Handling
+
+Rebar3 considers dependency versions to be informational only. Given the existing open source landscape in the Erlang community when rebar3 was added, trying to impose [semantic versioning](http://semver.org/) or any other similar scheme was considered unpractical:
+
+- People update *some* versions but not all of them (git tags vs. branch names vs. OTP application versions) and they may contradict each other;
+- Some people never update their versions and publish many times under them;
+- Not everyone subscribes to the same version schemes;
+- People make errors in subscribing to semantic versioning;
+- Many applications are stuck in versions smaller than `1.0.0`, and therefore considered unstable forever;
+- Source dependencies are frequently used: figuring out version conflicts therefore requires downloading all transitive dependencies from all dependencies to figure out whether they conflict or not, every time, and is costly
+- Strict adherence to semantic versioning ends up causing false conflicts where using a subset of an API which hasn't changed across major versions still requires manual conflict resolution and dependency audits.
+
+Instead, Rebar3 will fetch and download dependencies in a [level-order traversal](#fetching-order). This means that the dependencies closest to the root of the dependency tree are those that will be chosen, regardless of their versions. Any dependency declared in your project's `rebar.config` will never be overwritten by a transitive dependency, and a transitive dependency will never be overridden by a later encountered conflicting transitive dependency.
+
+This also means that if you prefer a version to be used above all else, you can just add it to your `rebar.config` file and pick what will be kept; the same conflict resolution mechanism sometimes required by semantic versioning.
+
+In practice, this has proven to be absolutely adequate as a mechanism.
+
+After each run of dependency fetching and resolving the list of final dependencies are written to `rebar.lock`.
+
+{{< blocks/callout type="info" title="Treating Conflicts as Errors">}}
+ If you ever want rebar3 to abort as soon as it detects a dependency conflict, instead of skipping the file and proceeding as usual, add the line <code>{deps_error_on_conflict, true}.</code> to your rebar configuration file.
+{{< /blocks/callout >}}
+
+For convenience reasons (and because [hex.pm](https://hex.pm) mandates semver) hex dependencies can be specified using semver-like syntax:
+
+```erlang
+{deps,[
+  rebar,                   % fetches latest known version, ignoring pre-releases
```

Comment:
Something I was starting to change else where is the use of `rebar` as a hex package in examples. I think better would be to use an actual package like `elli` with real version examples.

Reply:
Fair. I wouldn't block on it for this, but we should probably find something like that to work from.

I think when we wrote the initial docs we just didn't have a lot of packages with many versions to go around but we could use one now.

Reply:
Yea, agreed. I can take care of it when I do a full sweep of replacing `rebar` as the dep example.